pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -337,3 +375,52 @@ func gitRefInfo(repo git.Repository, dir, ref string) (buildapi.GitRefInfo, erro
 	info.Message = lines[4]
 	return info, nil
 }
+
+// WaitForBuildComplete waits for a build identified by the name to complete
+func WaitForBuildComplete(c osclient.BuildInterface, name string) error {
+	isOK := func(b *buildapi.Build) bool {
+		return b.Status.Phase == buildapi.BuildPhaseComplete
+	}
+	isFailed := func(b *buildapi.Build) bool {
```

Comment:
This is incosistent with what happens on the server where a failed build isn't considered an error: https://github.com/openshift/origin/blob/90750d8a2066c6c7469bef0419649d2c4d6555cc/pkg/build/registry/buildlog/rest.go#L136